ansible_builder/main.py

```diff
@@ -59,6 +64,63 @@ def __init__(self,
             galaxy_required_valid_signature_count=galaxy_required_valid_signature_count,
             galaxy_ignore_signature_status_codes=galaxy_ignore_signature_status_codes)
         self.verbosity = verbosity
+        self.container_policy, self.container_keyring = self._handle_image_validation_opts(container_policy, container_keyring)
+
+    def _handle_image_validation_opts(self, policy, keyring):
+        """
+        Process the container_policy and container_keyring arguments.
+
+        :param str policy: The container_policy value.
+        :param str keyring: The container_keyring value.
+
+        The container_policy and container_keyring arguments come from the CLI
+        and work together to help build or use a podman policy.json file used
+        do image validation. Depending on the policy being used, the keyring
+        may or may not be necessary.
+
+        The keyring, if required, must be a valid path, and will be transformed
+        to an absolute path to be used in the policy.json file.
+
+        :returns: A tuple of a PolicyChoices enum and abs path to the keyring.
+        """
+        resolved_policy = None
+        resolved_keyring = None
+
+        if policy is not None:
+            if self.version != "2":
+                raise ValueError(f'--container-policy not valid with version {self.version} format')
+
+            # Require podman runtime
+            if self.container_runtime != 'podman':
+                raise ValueError('--container-policy is only valid with the podman runtime')
+
+            resolved_policy = PolicyChoices(policy)
+
+            # Require keyring if we write a policy file
+            if resolved_policy == PolicyChoices.SIG_REQ and keyring is None:
+                raise ValueError(f'--container-policy={resolved_policy.value} requires --container-keyring')
+
+            # Do not allow images to be defined with --build-arg CLI option if
+            # any sig policy is defined.
+            for key, _ in self.build_args.items():
+                if key in ('EE_BASE_IMAGE', 'EE_BUILDER_IMAGE'):
+                    raise ValueError(f'{key} not allowed in --build-arg option with --container-policy={resolved_policy.value}')
```

Comment:
This exception message should probably not reference container policy since defining these images is invalid for version 2 format. Something more like:

```
EE_BASE_IMAGE not allowed to be defined with version 2 format
```